hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/TransactionHandler.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.spi.workflows;
+
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A {@code TransactionHandler} contains all methods for the different stages of a single operation.
+ */
+public interface TransactionHandler {
+
+    /**
+     * This method is called during the ingest-workflow. It does pre-checks that are specific to the
+     * operation.
+     *
+     * @param txBody the {@link TransactionBody} that needs to be validated
+     * @throws NullPointerException if {@code txBody} is {@code null}
+     * @throws PreCheckException if validation fails
+     */
+    void preCheck(@NonNull TransactionBody txBody) throws PreCheckException;
+}
```

Comment:
I wrote the `inheritDoc` comment because I assumed `handle` and `preHandle` were here as well. Do we not want them here?

Reply:
These methods will require different parameters depending on what is needed in the operation - at least initially. Later we will see, if we can generalize them.

Reply:
Is that so? I thought all the prehandles required only payer and txBody and can get everything else from the txBody.